Octokit.Reactive/Clients/IObservableTeamDiscussionsClient.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reactive;
+
+namespace Octokit.Reactive
+{
+    /// <summary>
+    /// A client for GitHub's Teams Discussions API.
+    /// </summary>
+    /// <remarks>
+    /// See the <a href="https://developer.github.com/v3/teams/discussions/">Team Discussions API documentation</a> for more information.
+    /// </remarks>
+    public interface IObservableTeamDiscussionsClient
+    {
+        /// <summary>
+        /// Get a specific discussion on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/teams/discussions/#get-a-single-discussion
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <param name="number">The team discussion number.</param>
+        /// <returns>The <see cref="TeamDiscussion"/> with the given identifier.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> Get(int teamId, int number);
+
+        /// <summary>
+        /// List all discussions on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/teams/discussions/#list-discussions
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <returns>A list of discussions in the team.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> GetAll(int teamId);
+
+        /// <summary>
+        /// List all discussions on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/teams/discussions/#list-discussions
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <param name="options">Options to change API behaviour.</param>
+        /// <returns>A list of discussions in the team.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> GetAll(int teamId, ApiOptions options);
+
+        /// <summary>
+        /// Creates a new discussion post on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// OAuth access tokens require the write:discussion scope.
+        /// https://developer.github.com/v3/teams/discussions/#create-a-discussion
+        /// </remarks>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        /// <returns>Newly created <see cref="TeamDiscussion"/> object.</returns>
+        IObservable<TeamDiscussion> Create(int teamId, NewTeamDiscussion teamDiscussion);
```

Comment:
can we please name the variable `NewTeamDiscussion newTeamDiscussion` here and in the implementation?

```suggestion
        IObservable<TeamDiscussion> Create(int teamId, NewTeamDiscussion newTeamDiscussion);
```